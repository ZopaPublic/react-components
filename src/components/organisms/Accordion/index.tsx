import AccordionComponent from './Accordion/Accordion';
import AccordionHeader from './AccordionHeader/AccordionHeader';
import AccordionSection from './AccordionSection/AccordionSection';

export const Accordion = Object.assign(AccordionComponent, {
  Header: AccordionHeader,
  Section: AccordionSection,
});
