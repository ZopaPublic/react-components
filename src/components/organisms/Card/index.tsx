import React, { HTMLAttributes } from 'react';

import CardActions from './CardActions/CardActions';
import CardComponent, { CardProps } from './Card/Card';
import CardContent from './CardContent/CardContent';
import CardHeading from './CardHeading/CardHeading';
import CardImage from './CardImage/CardImage';
import CardText from './CardText/CardText';
import CardLineItem from './CardLineItem/CardLineItem';

const Card = (props: CardProps & HTMLAttributes<HTMLDivElement>) => <CardComponent {...props} />;

export default Object.assign(Card, {
  Actions: CardActions,
  Content: CardContent,
  Heading: CardHeading,
  Image: CardImage,
  Text: CardText,
  LineItem: CardLineItem,
});
