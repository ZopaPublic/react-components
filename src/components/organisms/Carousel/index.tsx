import CarouselComponent from './Carousel/Carousel';
import Slide from './CarouselSlide/CarouselSlide';
import SlideIcon from './CarouselSlideIcon/CarouselSlideIcon';
import SlideText from './CarouselSlideText/CarouselSlideText';

export const Carousel = Object.assign(CarouselComponent, {
  Slide: Slide,
  SlideIcon: SlideIcon,
  SlideText: SlideText,
});
