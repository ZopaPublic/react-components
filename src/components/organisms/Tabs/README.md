### Summary

`<Tabs />` is a wrapper that is supposed to be used with `<Tabs.Buttons />` and `<Tabs.Content />`. It provides its children with the `<TabsContext />`.

### Example

```tsx
import { Tabs } from '@zopauk/react-components';

function TabsExample() {
  const tabButtons = [
    { tabId: 'pineapple', title: '🍍 Pineapple' },
    { tabId: 'kiwi', title: '🥝 Kiwi' },
    {
      tabId: 'watermelon',
      title: '🍉 Watermelon',
      afterOnClick: () => console.log('This tab has a function that is called after the default action in onClick'),
    },
  ];
  return (
    <Tabs>
      <Tabs.Buttons tabButtons={tabButtons} defaultTab="pineapple" />
      <Tabs.Content contentFor="pineapple">🍍</Tabs.Content>
      <Tabs.Content contentFor="kiwi">🥝</Tabs.Content>
      <Tabs.Content contentFor="watermelon">🍉</Tabs.Content>
    </Tabs>
  );
}

<TabsExample />;
```
