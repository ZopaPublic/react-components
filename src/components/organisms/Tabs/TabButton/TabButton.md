### Summary

`<TabsButton />` is rendered on wider screens and smaller screens if there are < 2 tabs.
