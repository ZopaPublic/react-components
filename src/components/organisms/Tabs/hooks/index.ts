export * from './useTabs';
export { TabsContext } from './TabsContext';
export { useTabsContext } from './useTabsContext';
