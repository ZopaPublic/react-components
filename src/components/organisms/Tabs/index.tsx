import React from 'react';

import TabsContainer, { TabsProps } from './TabsContainer/TabsContainer';
import TabContent from './TabContent/TabContent';
import TabButtons from './TabButtons/TabButtons';

const TabsMain = (props: TabsProps) => <TabsContainer {...props} />;

export const Tabs = Object.assign(TabsMain, { Content: TabContent, Buttons: TabButtons });

export * from './hooks';
