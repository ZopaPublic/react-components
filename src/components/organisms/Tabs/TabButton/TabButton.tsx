import React from 'react';
import styled, { css } from 'styled-components';
import { colors } from '../../../../constants';
import Button, { ButtonProps } from '../../../atoms/Button/Button';
import { useTabsContext } from '../hooks/useTabsContext';

export interface TabButtonProps extends ButtonProps {
  title: string;
  tabId: string;
  afterOnClick?: () => void;
}

interface StyledButtonProps {
  activeTab: string;
  tabId: string;
}

// This wrapper is to prevent html attribute warnings. See: https://styled-components.com/docs/faqs#why-am-i-getting-html-attribute-warnings
const ButtonWrapper = ({ activeTab, tabId, ...props }: StyledButtonProps & ButtonProps) => (
  <Button className="py-2" {...props} />
);

const StyledButton = styled(ButtonWrapper)`
  box-shadow: inset 0 12px ${colors.greyLightest}, inset 0 -12px ${colors.greyLightest}, inset 1px 0 ${colors.greyLight};
  flex-grow: 1;
  width: 100%;

  &:first-child {
    box-shadow: none;
  }

  &:hover,
  :focus {
    box-shadow: none;
    & + button {
      box-shadow: none;
    }
  }

  ${({ activeTab, tabId }) =>
    activeTab === tabId &&
    css`
      background: ${colors.white};
      box-shadow: none;

      & + button {
        box-shadow: none;
      }
    `}
`;

const TabButton = ({ title, tabId, afterOnClick }: TabButtonProps) => {
  const { activeTab, getTabButtonHTMLProps } = useTabsContext();
  const tabProps = getTabButtonHTMLProps(tabId, afterOnClick);

  return (
    <StyledButton activeTab={activeTab} tabId={tabId} styling="link" {...tabProps}>
      {title}
    </StyledButton>
  );
};

export default TabButton;
