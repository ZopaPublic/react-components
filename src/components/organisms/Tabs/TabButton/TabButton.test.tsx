import axe from '../../../../../axe-helper';
import React from 'react';
import { render } from '@testing-library/react';
import TabButton from './TabButton';
import { Tabs } from '../../../organisms/Tabs';

describe('<TabButton />', () => {
  it('renders the component with no a11y violations', async () => {
    const { container } = render(
      <Tabs>
        <TabButton tabId="pineapple" title="🍍 Pineapple" />
      </Tabs>,
    );
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
