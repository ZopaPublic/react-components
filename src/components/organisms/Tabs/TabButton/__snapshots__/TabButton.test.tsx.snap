// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabButton /> renders the component with no a11y violations 1`] = `
.c0 {
  outline: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 12px 24px;
  background: transparent;
  border-radius: 8px;
  border: 1px solid transparent;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  color: #21247F;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.c0:focus:not(:active) {
  border: 1px solid #FFFFFF;
  box-shadow: 0 0 4px #3B46C4;
}

.c0:active {
  opacity: 0.8;
  box-shadow: unset;
  border: 1px solid transparent;
}

.c0:hover {
  background: #EAEBFA;
}

.c1 {
  box-shadow: inset 0 12px #F7F7F7,inset 0 -12px #F7F7F7,inset 1px 0 #D4D7D9;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 100%;
}

.c1:first-child {
  box-shadow: none;
}

.c1:hover,
.c1:focus {
  box-shadow: none;
}

.c1:hover + button,
.c1:focus + button {
  box-shadow: none;
}

<div>
  <button
    aria-controls="pineapple-content"
    aria-expanded="false"
    class="c0 c1"
    id="pineapple"
    styling="link"
    theme="[object Object]"
  >
    üçç Pineapple
  </button>
</div>
`;
